### PR TITLE
[Quantum] Updates to enable resource estimation in the CLI

### PR DESCRIPTION
**Pull request by anjbur**
PR for az quantum extension, providing updates for resource estimation.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Quantum

File: src/quantum/azext_quantum/operations/job.py [hunk 1]
```diff
@@ -265,7 +270,7 @@ def output(cmd, job_id, resource_group_name=None, workspace_name=None, location=
         if len(lines) == 0:
             return
 
-        if job.target.startswith("microsoft.simulator"):
+        if job.target.startswith("microsoft.simulator") and job.target != "microsoft.simulator.resources-estimator":
```
Comment (msoeken) [inline src/quantum/azext_quantum/operations/job.py:273]: Is this the correct name?

Comment (shenyingjun) [reply src/quantum/azext_quantum/operations/job.py:273]: yes I think we changed the sku name, but not the target name

**Review by shenyingjun — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**